onnxruntime/core/providers/cuda/nn/pool.cc

```diff
@@ -152,7 +153,12 @@ Status Pool<T, PoolType, NHWC>::ComputeInternal(OpKernelContext* context) const
   const TensorShape& x_shape = X->Shape();
   const auto x_dims = x_shape.GetDims();
 
-  if (x_shape.NumDimensions() < 3) {
+  if constexpr (NHWC) {
+    if (kernel_shape.size() < 2)
+      return ORT_MAKE_STATUS(ONNXRUNTIME, NOT_IMPLEMENTED, "1D pooling for NHWC is not implemented yet");
+    if (x_shape.NumDimensions() < 4)
```

Comment:
Nit: Num input dims can be < 4 for 1-D Pool [N, feature_dim, C] ?

Reply:
Replied above: https://github.com/microsoft/onnxruntime/pull/19780#discussion_r1520511305